[Ha Jiwoon] iP

juneha1120 opened the pull request:
### ChillGuy

> "When your to-do list is full with overdue ones but you are just a chill guy." - Chill Guy

ChillGuy is here to just chill most of the time but also take care of things you need to do. It is,
- text-based
- easy to learn
- easy to use
- very chill

All you need to do is,
1. download it from [here](https://github.com/nus-cs2103-AY2425S2/ip/pull/86).
2. double-click it.
3. add your tasks.
4. let it chill 😎 

And it is **FREE**!

Features :
- [x] Managing tasks 
- [x] Finding tasks
- [x] Chilling
- [ ] Reminders (Coming soon!)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class ChillGuy {
    public static void main(String[] args) {
        new ChillGuy("./data/chillguy.txt").run();
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

juneha1120 committed:
Rename main class and remove all codes

juneha1120 committed:
Add greeting and exiting message

juneha1120 committed:
Add user input echoing feature

juneha1120 committed:
Separate Juno into two; Juno and JunoUI

juneha1120 committed:
Add help command and organise commands using map

juneha1120 committed:
Add task-adding command and feature to show added tasks

juneha1120 committed:
Add separate class for JunoTask

juneha1120 committed:
Add mark, unmark command to mark and unmark added tasks and add command to add task

juneha1120 committed:
Add markTask and unmarkTask to mark and unmark task and show appropriate message
Edit showTasks to show status of task as well

juneha1120 committed:
Add todo, event, deadline task types as classes

juneha1120 committed:
Edit JunoTask class so that todo, deadline, and event classes can inherit from it without issues

juneha1120 committed:
Edit JunoUI class to add and show task with different task types

juneha1120 committed:
Add Juno logo in JunoUI class

juneha1120 committed:
Edit input and EXPECTED output file to semi-automate the testing with text-ui-test

juneha1120 committed:
Delete last empty line in the codes

juneha1120 committed:
Add enum class for command types

juneha1120 committed:
Edit main Juno logic using enums

juneha1120 committed:
Add JunoException to create exceptions according to command types and specifications

juneha1120 committed:
Add JunoTasks to separately handle the list of tasks in JunoUI and throw JunoExceptions
Edit JunoUI to handle JunoExceptions that can be caught

juneha1120 committed:
Edit input and EXPECTED output to account for changes

juneha1120 committed:
Edit JunoTasks to handle the logic in deleteTask
Edit JunoUI to print appropriate message when deleteTask is called

juneha1120 committed:
Edit main Juno logic to simplify input choices

juneha1120 committed:
Edit JunoType enum to add DELETE as a new command type

juneha1120 committed:
Edit input and EXPECTED output to account for changes in adding delete feature in Level-6

juneha1120 committed:
Add tasks renumbering method in JunoTasks to account for changes in numbering of task list when tasks are deleted

juneha1120 committed:
Edit input and EXPECTED output to consider deleting and renumbering feature

juneha1120 committed:
Edit JunoTasks, JunoTask and JunoExceptions so that marking already marked task as well as unmarking already unmarked task follow different logic
Edit JunoUI to output appropriate message when marking marked task and unmarking unmarked task

juneha1120 committed:
Edit input and EXPECTED output to account for changes made in previous commit

juneha1120 committed:
Edit codes to follow code standard

juneha1120 committed:
Remove enum JunoType.java
Add enum JunoTaskType.java and enum JunoErrorType.java

Single enum JunoType is used for referencing all types of command and
names in Juno

Not so sensible to use common JunoType enums for referencing both tasks
and errors using enums

Separating into enum JunoTaskType and JunoErrorType to handle types of
tasks and errors separately

juneha1120 committed:
JunoUI.java : Add loadTasks() and add IOException catching

No supported UI for task loading and no catching IOException that can be
thrown from task saving

Add loadTasks() to print relevant messages and call loadTasks() in
JunoTasks.java followed by showTasks() and catching relevant exceptions
Add IOException catching for methods that calls saveTasks() in JunoTasks

juneha1120 committed:
Add toFileFormat() method for JunoTodo, JunoDeadline, JunoEvent

They do not have method to yield appropriate format for saving

To save each task in text file, requires toFileFormat() method

Add toFileFormat() method to get task type, status and its description
as string supported for saving

juneha1120 committed:
Add saveTasks() and loadTasks() method
Add juno.txt under data
Edit JunoException throwing details for each method

Juno does not support saving current task list and loading it from
previous use

Add saveTasks() and loadTasks() to save task list under ./data/juno.txt
and load it from it while handling relevant exceptions

juneha1120 committed:
Add overloaded getErrorMessage methods

Single getErrorMessage method deals with all errors with different
conditions

Add four different getErrorMessage method to deal with respective errors
with different conditions

juneha1120 committed:
Add abstract toFileFormat() method and fromFileFormat method

Juno does not support retrieving appropriate task from saved text file

Add abstract toFileFormat() for subclasses to implement and fromFileFormat
method to retrieve task and return it to JunoTasks by line

juneha1120 committed:
Add ui.loadTasks() to load saved tasks when starting

juneha1120 committed:
Merge branch 'branch-Level-7'

juneha1120 committed:
Juno.java : Add ui.showTasksOnDate()

juneha1120 committed:
JunoDeadline.java : Change String by to LocalDateTime by and Add LocalDate byDate
JunoEvent.java : Change String from and to to LocalDateTime from and to and
				   Add LocalDate fromDate and toDate

juneha1120 committed:
JunoException.java : Add LIST_WITH_DATE_ERROR and Edit error message for
 					   DEADLINE_ERROR and EVENT_ERROR

juneha1120 committed:
JunoTask.java : Edit fromFileFormat() to handle LocalDate and LocalDateTime

juneha1120 committed:
JunoErrorType.java : Add LIST_WITH_DATE_ERROR

juneha1120 committed:
JunoTasks.java : Edit makeTask() to handle LocalDate/LocalDateTime and
				   Edit showTasks() to getTasks() and
				   Add getTasksOnDate() to get List<JunoTask>

juneha1120 committed:
JunoUI.java : Add showTasksOnDate method

juneha1120 committed:
input.txt : Edit to account for Level-7 and Level-8
EXPECTED.TXT : Edit to account for Level-7 and Level-8

juneha1120 committed:
Merge branch 'branch-Level-8'

juneha1120 committed:
Remove oriniginal Juno files to employ new OOP

juneha1120 committed:
Command.java : Add abstract execute method

juneha1120 committed:
AddCommand.java : Add constructor with Task and execute method

juneha1120 committed:
DeleteCommand.java : Add constructor with taskNum, deleteTask method,
renumber Tasks method, execute method

juneha1120 committed:
ExitCommand.java : Add isExit method and execute method

juneha1120 committed:
HelpCommand.java : Add execute method

juneha1120 committed:
MarkCommand.java : Add execute method

juneha1120 committed:
ShowTasksCommand.java : Add execute method

juneha1120 committed:
ShowTasksWithDateCommand.java : Add constructor with LocalDate,
getTasksOnDate method and execute method

juneha1120 committed:
TestCommand.java : Add execute method to just ignore

juneha1120 committed:
TryAgainCommand.java : Add execute method

juneha1120 committed:
UnmarkCommand.java : Add constructor with taskNum and execute method

juneha1120 committed:
CommandList.java : Add commandDescriptionList that contains string array
of all the tasks' description

juneha1120 committed:
Rename JunoTaskType.java to TaskType.java

juneha1120 committed:
Rename JunoErrorType.java to ErrorType.java
Add few more error types

juneha1120 committed:
JunoException.java : Rename few error messages and Move to new directory
JunoTestException.java : Add constructor which does not have message

juneha1120 committed:
Ui.java : Add readCommand and showToUser command along with necessary
strings and other show commands for different juno commands

juneha1120 committed:
Messages.java : Add string messages needed for Ui.java

juneha1120 committed:
Parser.java : Add parse method along with helper methods to parse string
inputs

juneha1120 committed:
Storage.java : Add fromFileFormat method to extract Task, saveTasks to
filePath and loadTasks from filePath methods

juneha1120 committed:
Task.java : Move to new directory

juneha1120 committed:
Move Todo.java, Deadline.java, and Event.java to new directory and
add missing constructors along with helper strings

juneha1120 committed:
TaskList.java : Move to new directory and add getStringTaskList method
that returns string array of task list

juneha1120 committed:
Juno.java : Edit main logic of Juno to account for changes

juneha1120 committed:
input.txt : Add commenting lines to better understand inputs
EXPECTED.TXT : Add corresponding expected outputs

juneha1120 committed:
Merge branch 'branch-A-MoreOOP'

juneha1120 committed:
Move CommandList.java, Command.java and its subclasses to juno/command/

juneha1120 committed:
Move ErrorType.java and TaskType.java to juno/enums/

juneha1120 committed:
Rename juno/command to juno/commands

juneha1120 committed:
Rename juno.command to juno.commands

juneha1120 committed:
Move JunoException.java and JunoTestException.java to juno/exceptions/

juneha1120 committed:
Move Parser.java to juno/parser/

juneha1120 committed:
Move Juno.java and Messages.java to juno/main/

juneha1120 committed:
Move TaskList.java, Task.java and its subclasses to juno/task/

juneha1120 committed:
Move Storage.java to juno/ui/

juneha1120 committed:
Move Ui.java to juno/ui/

juneha1120 committed:
HelpCommand.java : Revert COMMAND_WORD string that was
accidentally changed while refactoring

juneha1120 committed:
Merge branch 'branch-A-Packages'

juneha1120 committed:
Accessory commits

juneha1120 committed:
Merge branch 'add-gradle-support'

juneha1120 committed:
Parser.java : Edit access modifiers to protected for ParserTest to
access them

juneha1120 committed:
Remove redundant second argument which can be extracted from fields for
getTasksOnDate in ShowTasksWithDateCommand.java and deleteTask in
DeleteCommand.java

juneha1120 committed:
Edit code wording that uses description and getDescription() to
taskName and getTaskName() for clarity

juneha1120 committed:
Storage.java : Add EXAMPLE filePath

juneha1120 committed:
Add CommandListTest.java, CommandTest.java and its subclasses tests

juneha1120 committed:
Add TaskListTest.java, TaskTest.java and its subclasses tests

juneha1120 committed:
Add UiTest.java

juneha1120 committed:
Add StorageTest.java

juneha1120 committed:
Add ParserTest.java

juneha1120 committed:
Add JunoTest.java

juneha1120 committed:
Merge branch 'branch-A-JUnit'

juneha1120 committed:
Tweak CommandList.java, Command.java and its subclasses to comply with
coding standard

juneha1120 committed:
Tweak TaskType.java and ErrorType.java to comply with coding standard

juneha1120 committed:
Tweak JunoException.java and JunoTestException.java to comply with coding
standard

juneha1120 committed:
Tweak Juno.java and Messages.java to comply with coding standard

juneha1120 committed:
Tweak Parser.java to comply with coding standard

juneha1120 committed:
Tweak Storage.java to comply with coding standard

juneha1120 committed:
Tweak TaskList.java, Task.java and its subclasses to comply with
coding standard

juneha1120 committed:
Tweak Ui.java to comply with coding standard

juneha1120 committed:
Tweak all test java files to comply with coding standard

juneha1120 committed:
FindCommand.java : Add constructor with keyword, getTasksWithKeyword and
execute method

juneha1120 committed:
CommandList.java : Add FindCommand command description in the list

juneha1120 committed:
JunoException.java : Add cases for exceptions thrown from FindCommand
ErrorType.java : Add enums for error types thrown from FindCommand

juneha1120 committed:
Messages.java : Add starting message for find command

juneha1120 committed:
Parser.java : Add prepareFindCommand method and case for FindCommand
in main parse logic

juneha1120 committed:
Ui.java : Add showFind method to print relevant infos when finding

juneha1120 committed:
Add FindCommandTest junit

juneha1120 committed:
input.txt : Add inputs for testing find command
EXPECTED.TXT : Add corresponding outputs for find command

juneha1120 committed:
Merge branch 'branch-A-CodingStandard'

juneha1120 committed:
Merge Javadoc conflicts

juneha1120 committed:
Add Javadoc for java files

juneha1120 committed:
Merge branch 'master' into branch-A-JavaDoc

juneha1120 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/juno/commands/CommandList.java
#	src/main/java/juno/enums/ErrorType.java
#	src/main/java/juno/exceptions/JunoException.java
#	src/main/java/juno/parser/Parser.java
#	src/main/java/juno/ui/Ui.java

juneha1120 committed:
Rename Juno to ChillGuy

juneha1120 committed:
Edit messages and exception messages to fit ChillGuy language

juneha1120 committed:
Use CheckStyle

juneha1120 committed:
Merge branch 'branch-A-CheckStyle'

juneha1120 committed:
Add GUI to ChillGuy

juneha1120 committed:
Rename Ui to TextUi

juneha1120 committed:
Merge branch 'branch-Level-10'

ambertan77 commented on src/main/java/chillguy/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,110 @@
+package chillguy.commands;
+
+import static chillguy.enums.ErrorType.DELETE_ERROR;
```
Should this import statement be placed together with the rest of the imports from the `chillguy` package? I've also noticed a similar pattern in other files, such as in the `Storage.java` and `Parser.java` files.

juneha1120 replied:
Thank you for the comment! Using checkstyle.xml, although it is in chillguy package, it requires static import like this to be put on top of all other imports if i remember correctly.

ambertan77 commented on src/main/java/chillguy/exceptions/ChillGuyException.java:
```diff
@@ -0,0 +1,137 @@
+package chillguy.exceptions;
```
I like how you have created a new Exception to give more details about any error that may have occured :)

juneha1120 replied:
Thank you!

ambertan77 commented on src/main/java/chillguy/enums/ErrorType.java:
```diff
@@ -0,0 +1,14 @@
+ * <p>
+ * The {@code ErrorType} enum defines various error types that the system may encounter:
+ */
+public enum ErrorType {
```
Not sure if it would be better to have the same prefix for these constants? Since enums represent a set of constants and all the constants defined here are associated (all are error types).

juneha1120 replied:
Thank you for the comment! But I am not entirely clear if you meant javadoc comments as in prefix here?

AerollingOverDeepBlueSeas replied:
I think the previous reviewer may have meant the names of the `enum`. For instance, instead of naming them `CREATE_FILE_ERROR` and `DATE_ERROR`, they were suggesting to rename them to `ERROR_CREATE_FILE` and `ERROR_DATE`. They may have been referring to the guideline "Associated constants should have a common prefix." from the Java coding standard.
(I personally think the guideline "Related things should be named similarly, while unrelated things should NOT." from [Coding Quality: Naming](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/topics.html#w4-6-code-quality-naming) would be more applicable here, and that your current naming convention works alright.)

ambertan77 commented on src/main/java/chillguy/exceptions/ChillGuyException.java:
```diff
@@ -0,0 +1,137 @@
+        case LIST_WITH_DATE_ERROR -> "We chill, but looks like you have no tasks on the date.";
+        case LIST_WITH_KEYWORD_ERROR -> "We chill, but looks like you have no matching ones.";
+        case NO_KEYWORD_ERROR -> "We chill, but could you tell me the keyword to search?";
+        default -> ""; };
```
Should the closing curly bracket `}` here and in line 98 be on the next line (the line after the default case)?

juneha1120 replied:
Thank you for the comment. That is right I will edit accordingly!

ambertan77 commented on src/main/java/chillguy/main/ChillGuy.java:
```diff
@@ -0,0 +1,128 @@
+                isExit = ExitCommand.isExit(c);
+                this.textUi.showDivider();
+            } catch (ChillGuyTestException ignored) {
+                // Ignored
```
May I ask why the `ChillGuyTestException` here needs to be caught? Would it not be caught by the catch block which catches the `ChillGuyException`?

juneha1120 replied:
Thank you for the comment! Actually for ChillGuyTestException, when it gets caught, it should not do anything unlike other exceptions so I decided to put it above ChillGuyException catching line. If you look at my input.txt file under text-ui-test directory, it might be a bit more clear

ambertan77 reviewed (commented):
Overall, I think your code is very clean and easy to read and follow! Nice job :)

juneha1120 committed:
Edit for cleaner code and better presentation

juneha1120 committed:
Change GUI features

juneha1120 committed:
Use assert feature to document assumptions

Throughout the codebase, there are several points where assumptions about
the program's state need to be explicitly checked to ensure consistency,
correctness, and data integrity.

Using Java’s assert feature allows us to efficiently document and
enforce these assumptions.

Let's add assertions in critical methods (e.g., task addition/removal,
data validation) to check that the task list’s size corresponds with the
number of tasks, ensuring no tasks are lost or duplicated.

These changes improve code quality by documenting important invariants
that should always hold true, providing a quick way to catch logical
inconsistencies during runtime.

juneha1120 committed:
Improve code quality

The current code has areas that can benefit from refactoring to improve
readability, maintainability, and overall quality.

Let's:
* refactor methods that are too large or complex into smaller, more
focused methods
* improve naming conventions for variables and methods to increase
clarity

This improves code quality by catching errors and documenting important
invariants.

juneha1120 committed:
Use Streams for task list conversion

The getStringTaskList() method previously used a manual iteration over
the taskList map, adding formatted task strings to an ArrayList, and
then converting it to an array.

This approach can be simplified and made more efficient using the
Streams API, which provides a cleaner and more declarative way to
perform the transformation.

Let's refactor the method to use the Stream API for processing the
taskList map.

This change improves code readability, reduces verbosity, and leverages
Java's functional programming features to make the code more concise
and maintainable.

juneha1120 committed:
Merge pull request #1 from juneha1120/branch-A-Assertions

Use assert feature to document assumptions

juneha1120 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/chillguy/commands/ExitCommand.java
#	src/main/java/chillguy/commands/ShowTasksCommand.java
#	src/main/java/chillguy/commands/TryAgainCommand.java

juneha1120 committed:
Merge branch 'master' into branch-A-Streams

juneha1120 committed:
Merge pull request #2 from juneha1120/branch-A-CodeQuality

Improve code quality

juneha1120 committed:
Merge branch 'master' into branch-A-Streams

juneha1120 committed:
Merge pull request #3 from juneha1120/branch-A-Streams

Use Streams for task list conversion

juneha1120 committed:
Set up CI

juneha1120 committed:
Provide reminders for the current date

juneha1120 committed:
Merge branch 'branch-B-Reminders'

juneha1120 committed:
Add minor tweaks for better presentation and logic flow

juneha1120 committed:
Update text-ui-test

juneha1120 committed:
Add minor tweaks for junit testing

juneha1120 committed:
Add minor tweaks in task string format

juneha1120 committed:
Edit gradle.yml

juneha1120 committed:
Edit gradle.yml again

juneha1120 committed:
Edit runtest.bat for it to work on github workflow

AerollingOverDeepBlueSeas commented on src/main/java/chillguy/commands/CommandList.java:
```diff
@@ -0,0 +1,40 @@
+        EMPTY_DIVIDER,
+        LINE_PREFIX + DeleteCommand.COMMAND_DESCRIPTION,
+        EMPTY_DIVIDER,
+        ExitCommand.getCommandDescription()
```
Why is the command description for `ExitCommand` retrieved via a method when all the other commands have a declared constant for the command description?

juneha1120 replied:
Thank you for the comment! As I implemented GUI, I decided to remove ExitCommand's description from the help command list and I did that through setting the description as empty when execute() is run with GUI. And to do that, I had to remove ExitCommand's COMMAND_DESCRIPTION's "final" access modifier and for some reason, I decided to change it to protected as well with getter. But I think I can just keep as public and just call ExitCommand.commandDescription

AerollingOverDeepBlueSeas commented on src/main/java/chillguy/parser/Parser.java:
```diff
@@ -0,0 +1,279 @@
+        case ExitCommand.COMMAND_WORD -> new ExitCommand();
+        case TestCommand.COMMAND_WORD, TestCommand.COMMAND_LINE, TestCommand.EMPTY_LINE ->
+                throw new ChillGuyTestException();
+        default -> new TryAgainCommand(); };
```
I think the closing `};` of the switch-case should be on a new line?

juneha1120 replied:
Thank you for the moment. That I think is absolutely right.. thank you for the correction 😄

AerollingOverDeepBlueSeas commented on src/main/java/chillguy/ui/TextUi.java:
```diff
@@ -0,0 +1,299 @@
+    public static final String EMPTY_DIVIDER = "";
+    public static final String LINE_PREFIX = " ";
+    public static final String[] CHILLGUY_LOGO = {
+        "                                      ░▒▒  ░▒▒                          ",
```
Woah that's the Chillguy logo 
(Sorry, saw this logo and I had to make a comment - it's cool:)

juneha1120 replied:
Thank you for the comment! It was something I just had to make for sure..😀

AerollingOverDeepBlueSeas reviewed (commented):
I really like the structure of your code! Especially the splitting of all the commands into classes of their own :O

juneha1120 committed:
Edit HelpCommand to split help for each command

juneha1120 committed:
Add minor tweaks in ui messages and GUI

juneha1120 committed:
Add a user guide with representative screenshot

juneha1120 committed:
Merge branch 'branch-A-UserGuide'

juneha1120 committed:
Add minor changes to README.md

juneha1120 committed:
Add minor changes to README.md

juneha1120 committed:
Add minor tweaks to user guide mark down file

juneha1120 committed:
Add minor changes to UserGuide.md

juneha1120 committed:
Rename back to README.md again

juneha1120 committed:
Update READ.md with new features

juneha1120 committed:
Edit README.md to handle errors

juneha1120 committed:
Edit README.md with emojis

juneha1120 committed:
Update README.md

juneha1120 committed:
Update README.md

juneha1120 committed:
Update README.md

juneha1120 committed:
Update README.md

juneha1120 committed:
Update README.md

juneha1120 committed:
Update README.md

juneha1120 committed:
Edit fxml files to fix javafx version mismatch warnings

juneha1120 committed:
Edit to enable mouse scroll without clickling on scrollbar

juneha1120 committed:
Add author tag credit to the original author for fixing scrolls

juneha1120 committed:
Edit mainwindow to clear input box as soon as it is saved for processing

juneha1120 committed:
Update README.md

juneha1120 committed:
Update README.md

juneha1120 committed:
Update README.md
